test/pkg/integration/setup.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package integration
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/e2e-framework/klient"
+	"sigs.k8s.io/e2e-framework/klient/conf"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+func Setup(ctx context.Context) (string, string, env.Environment) {
+	var (
+		namespace = flag.String("namespace", "default", `namespace to execute the test against. Defaults to the one configured in "kubeconfig".`)
+		username  = flag.String("username", "", "username to execute the tests with. Chooses one automatically if left blank.")
+	)
+
+	klog.InitFlags(nil)
+	flag.Parse()
+
+	restConfig, err := conf.New(conf.ResolveKubeConfigFile())
+	if err != nil {
+		klog.Fatalf("unexpected error: %v", err)
+	}
+
+	// change defaults to avoid limiting connections
+	restConfig.QPS = 20
+	restConfig.Burst = 50
+
+	client, err := klient.New(restConfig)
+	if err != nil {
+		klog.Fatalf("unexpected error: %v", err)
+	}
+
+	conf := envconf.New()
+	conf.WithClient(client)
+	conf.WithNamespace(*namespace)
+
+	ctx = context.WithValue(ctx, "username", *username)
+
+	testenv, err := env.NewWithContext(ctx, conf)
+	if err != nil {
+		klog.Fatalf("unexpected error: %v", err)
+	}
+	testenv.Setup(
+		checkGitpodIsRunning(*namespace),
+	)
+
+	return *username, *namespace, testenv
+}
+
+func checkGitpodIsRunning(namespace string) env.Func {
```

Comment:
/nit

-> waitOnGitpodRunning ?

Maybe make the Poll timeouts configureable?

Reply:
Awesome to have this functionality. Thought: it might make sense to move this into `gpctl` in the future, as it would be cool to have this in a couple of places to detect "end of deployment", for instance.

Reply:
done